lens/private/list/flatten.scrbl

```diff
@@ -0,0 +1,63 @@
+#lang scribble/manual
+
+@(require "../doc-util/main.rkt")
+
+@title{Flattening and unflattening lists}
+
+@defthing[append*-lens lens?]{
+A lens that flattens a list one-level down when viewing, and restores
```

Comment:
I'd add prose and examples describing what happens when the input is already a flat list, or when it's a single element.